ledger/participant-integration-api/src/main/scala/platform/store/dao/events/SqlFunctions.scala

```diff
@@ -41,5 +74,51 @@ private[dao] object SqlFunctions {
 
     def arrayIntersectionValues(arrayColumn: String, parties: Set[Party]): String =
       s"array_intersection($arrayColumn, array[${format(parties)}])"
+
+    def toArray(value: String) = s"array['$value']"
+
+    def limitClause(numRows: Int) = s"limit $numRows"
+
+    def equalsClause(left: String) = s"$left = "
+
+    def equalsClauseEnd(): String = ""
+
+    def greaterThanClause(left: String, right: Any) = s"$left > $right"
+
+    def greaterThanClauseStart(left: String): String = s"$left > "
+
+    def greaterThanClauseEnd(): String = ""
+
+    def lessThanOrEqualToClause(left: String, right: Any) = s"$left <= $right"
   }
+
+  //TODO need to properly implement this for Oracle
+  object OracleSqlFunctions extends SqlFunctions {
+    //TODO BH: this is likely extremely inefficient
+    override def arrayIntersectionWhereClause(arrayColumn: String, parties: Set[Party]): String =
+      parties
+        .map(party => s"('$party') IN (SELECT * FROM TABLE($arrayColumn))")
+        .mkString("(", " or ", ")")
+
+    def arrayIntersectionValues(arrayColumn: String, parties: Set[Party]): String =
+      s"CAST(MULTISET(select unique $arrayColumn.* FROM TABLE($arrayColumn) $arrayColumn intersect select * from TABLE(VARCHAR_ARRAY(${format(parties)}))) as VARCHAR_ARRAY)"
+
+    def toArray(value: String) = s"VARCHAR_ARRAY('$value')"
+
+    def limitClause(numRows: Int) = s"fetch next $numRows rows only"
+
+    def equalsClause(left: String) = s"DBMS_LOB.compare($left, "
```

Comment:
Please remove all functions and their usages that are related to LOB comparison.

Reply:
addressed